### ISSUE TITLE
Track unique objects

### ISSUE THREAD

**Issue by alex-savin** [blakeblackshear/frigate#476]
**Describe the bug**
Is there any way to track the same unique object.
I meant how it is possible to avoid of being notified about the same unique object, like a parked car or a still person.
Right now, I'm being bombarded by new messages about the same objects

**Version of frigate**
v0.8.0b3

**Config file**
Include your full config file wrapped in triple back ticks.
```yaml
---
logger:
  default: info
  logs:
    frigate.mqtt: error

database:
  path: /media/frigate/clips/frigate.db

detectors:
  coral:
    type: edgetpu
    device: pci

model:
  height: 320
  width: 320

mqtt:
  host: 10.XX.XX.XX
  topic_prefix: frigate
  client_id: frigate
  user: frigate
  password: REDACTED

save_clips:
  max_seconds: 60
  tmpfs_cache_size: 256m
  retain:
    default: 10
    objects:
      person: 15

ffmpeg:
  global_args: -hide_banner -loglevel fatal
  hwaccel_args: []
  input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclock_as_timestamps 1
  output_args:
    detect: -f rawvideo -pix_fmt yuv420p
    record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    clips: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    rtmp: -c copy -f flv

#objects:
#  track:
#    - person
#    - car
#    - truck
#  filters:
#    person:
#      min_area: 5000
#      max_area: 100000
#      threshold: 0.8

motion:
  threshold: 25
  contour_area: 100
  delta_alpha: 0.2
  frame_alpha: 0.2
  frame_height: 180

detect:
  max_disappeared: 10

cameras:
  frontyard:
    ffmpeg:
      inputs:
        - path: rtsp://10.XX.XX.XX:7447/XXXXXXXXXXXXXXX
          roles:
            - detect
            - rtmp
    height: 1080
    width: 1920
    fps: 30
    best_image_timeout: 60
    zones:
      zone_driveway:
        coordinates:
          - 141,1077
          - 0,1077
          - 0,395
          - 343,193
          - 1011,256
        filters:
          car:
            min_area: 5000
            max_area: 500000
            threshold: 0.7
      zone_property:
        coordinates:
          - 1915,1073
          - 0,1073
          - 7,168
          - 1913,346
        filters:
          person:
            min_area: 2000
            max_area: 150000
            threshold: 0.7
          cat:
            min_area: 1000
            max_area: 150000
            threshold: 0.7
    objects:
      track:
        - person
        - car
    save_clips:
      enabled: True
      pre_capture: 15
      objects:
        - person
    snapshots:
      show_timestamp: False
      draw_zones: True
      draw_bounding_boxes: True
```

**Logs**
```
 * Starting nginx nginx
   ...done.
frigate.app                    INFO    : Creating tmpfs of size 256m
frigate.app                    WARNING : Camera frontyard has save_clips enabled, but clips is not assigned to an input.
detector.coral                 INFO    : Starting detection process: 78
frigate.edgetpu                INFO    : Attempting to load TPU as pci
frigate.app                    INFO    : Camera processor started for frontyard: 81
frigate.app                    INFO    : Camera processor started for doorbell: 82
frigate.app                    INFO    : Camera processor started for entrance: 83
frigate.app                    INFO    : Capture process started for frontyard: 84
frigate.app                    INFO    : Capture process started for doorbell: 86
frigate.app                    INFO    : Capture process started for entrance: 90
frigate.edgetpu                INFO    : TPU found
frigate.video                  INFO    : doorbell: ffmpeg sent a broken frame. memoryview assignment: lvalue and rvalue have different structures
frigate.video                  INFO    : doorbell: ffmpeg process is not running. exiting capture thread...
```

**Frigate debug stats**
```
{"detection_fps":77.4,"detectors":{"coral":{"detection_start":1609707287.757457,"inference_speed":10.35,"pid":78}},"doorbell":{"camera_fps":30.0,"capture_pid":86,"detection_fps":31.2,"pid":82,"process_fps":29.8,"skipped_fps":0.0},"entrance":{"camera_fps":30.0,"capture_pid":90,"detection_fps":0.0,"pid":83,"process_fps":30.0,"skipped_fps":0.0},"frontyard":{"camera_fps":30.1,"capture_pid":84,"detection_fps":46.2,"pid":81,"process_fps":9.1,"skipped_fps":0.0}}
```

**FFprobe from your camera**

Run the following command and paste output below
```
ffprobe version 4.3.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-libfreetype --enable-libvidstab --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libxcb --enable-libx265 --enable-libxvid --enable-libx264 --enable-nonfree --enable-openssl --enable-libfdk_aac --enable-postproc --enable-small --enable-version3 --enable-libzmq --extra-libs=-ldl --prefix=/opt/ffmpeg --enable-libopenjpeg --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-libsrt --enable-libaribb24 --enable-nvenc --enable-cuda --enable-cuvid --enable-libnpp --extra-cflags='-I/opt/ffmpeg/include -I/opt/ffmpeg/include/ffnvcodec -I/usr/local/cuda/include/' --extra-ldflags='-L/opt/ffmpeg/lib -L/usr/local/cuda/lib64 -L/usr/local/cuda/lib32/'
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
[rtsp @ 0x558ba6b8cbc0] UDP timeout, retrying with TCP
[rtsp @ 0x558ba6b8cbc0] Could not find codec parameters for stream 1 (Video: h264, none): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
Input #0, rtsp, from 'rtsp://10.XX.XX.XX:7447/XXXXXXXXXXX':
  Metadata:
    title           : FCXXDAXX80XX_0
  Duration: N/A, bitrate: N/A
    Stream #0:0: Audio: aac, 11025 Hz, mono, fltp
    Stream #0:1: Video: h264, none, 90k tbr, 90k tbn, 180k tbc
```

**Screenshots**

**Computer Hardware**
 - OS: Debian 10
 - Virtualization: Proxmox
 - Coral Version: PCIe
 - Network Setup: Wired

**Camera Info:**
 - Manufacturer: Ubiquiti
 - Model: G3 Pro
 - Resolution: 1080
 - FPS: 30

**Additional context**

**Comment by blakeblackshear** [comment]
Is the `id` changing in the `frigate/events` mqtt topic? If so, that means frigate is losing track of the object and picking it up again. If not, they you are just getting notified when a better or newer version of that image is available.

If frigate is losing the object, I would try reducing the `fps` setting on your camera in frigate's config to 5 and increasing the `max_disappeared` setting. At 30fps, if frigate loses it for 1/3 of a second, it will mark it as lost and a new object id will be used when it is detected again.

**Comment by Halytskyi** [comment]
The same issue for me (fps 5, id not changing). Once person detected it sends 5-7 images within 1 seconds. Would be good to throttle or decrease number of images, for example, one image when object detected and 2nd one with the best snapshot.

**Comment by Halytskyi** [comment]
Maybe will be useful next scenario (as option and configurable): send 1st message once event start and send the 2nd messages when event finished (with the best detected snapshot).
P.S. I use "Telegram" for notification and even if it's possible set some limit on these notifications for "Telegram" (didn't check) I can lose some useful notification, therefore, not sure if it's will be good option...

**Comment by blakeblackshear** [comment]
I assume you are also using homeassistant? You can use conditions to only send the telegram notification on start and end. Would that work?

**Comment by Halytskyi** [comment]
Yes, tried with conditions, but still not find the good one... by which criteria I can know that event start and not continue..? For the end I can use "end_time" in the condition and check if it's not null, but it's also not working (seems I'm wrong with syntax), tried:
`
- "{{ trigger.payload_json['after']['end_time'] }}"
- "{{ trigger.payload_json['after']['end_time'] }} != 'null' "
- "{{ trigger.payload_json['after']['end_time'] }} != 'None' "
`

**Comment by blakeblackshear** [comment]
This should only allow the first event message and the last event message for each event id through:

```yaml
- condition: or
  conditions:
    - "{{ trigger.payload_json['after']['end_time'] is defined}}"
    - "{{ trigger.payload_json['before']['false_positive'] }}"
```

**Comment by Halytskyi** [comment]
nope... just tried that.. still about 3-5 images per second

**Comment by InToSSH** [comment]
I had the same issue. 
I am using Node-Red in Home Assistant to detect event updates, then there is a 5s trigger which is waiting for a snapshot with better score to arrive (I am not using the "end of event" report from Frigate, because some events take too long and I want to be notified earlier - before the end of event), so I am waiting 5sec for better score, if not, it will send a picture to my Telegram chat only once for each event (checks via rbe). I also send "inline keyboard" buttons with the picture so with single click on the button in the chat I can request a video clip, which is then sent as a new message to the chat.

If anyone is interested I can post a flow export.

btw I am also using 5FPS and max_disappeared set to 25 (5sec).

**Comment by saket424** [comment]
@InToSSH 
Please post your flow here with tripe backticks
Thanks

**Comment by InToSSH** [comment]
Here it is:
```
[{"id":"705e020d.9cfb5c","type":"tab","label":"Frigate (export)","disabled":false,"info":""},{"id":"dbf4ad89.a3621","type":"mqtt in","z":"705e020d.9cfb5c","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"a86136e0.7fe238","x":110,"y":180,"wires":[["578a71f8.e98cc"]]},{"id":"578a71f8.e98cc","type":"json","z":"705e020d.9cfb5c","name":"","property":"payload","action":"","pretty":false,"x":250,"y":180,"wires":[["f2e5bfee.808d2"]]},{"id":"b0c7e279.c2286","type":"api-call-service","z":"705e020d.9cfb5c","name":"","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"send_photo","entityId":"","data":"{\"target\":1111100000,\"url\":\"http://10.10.1.2:23500/events/{{payload.after.id}}/snapshot.jpg\",\"caption\":\"{{payload.after.label}} detected at {{payload.after.camera}}\",\"inline_keyboard\":[\"Get video:/fgv:{{payload.after.camera}}-{{payload.after.id}}, Siren short:/sirenshort, Siren:/siren\"]}","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":2350,"y":220,"wires":[[]]},{"id":"65c8ad05.a4d164","type":"switch","z":"705e020d.9cfb5c","name":"If score >= 0.6","property":"payload.after.score","propertyType":"msg","rules":[{"t":"gte","v":"0.6","vt":"str"}],"checkall":"true","repair":false,"outputs":1,"x":580,"y":180,"wires":[["38f13386.6609ec"]]},{"id":"31083dc8.dd2812","type":"rbe","z":"705e020d.9cfb5c","name":"Only once per event","func":"rbe","gap":"","start":"","inout":"out","property":"topic","x":1640,"y":220,"wires":[["6ed0f01a.1a549"]]},{"id":"8db8cd9.a79103","type":"trigger","z":"705e020d.9cfb5c","name":"","op1":"","op2":"","op1type":"nul","op2type":"payl","duration":"5","extend":true,"overrideDelay":false,"units":"s","reset":"","bytopic":"topic","topic":"topic","outputs":1,"x":1460,"y":220,"wires":[["31083dc8.dd2812"]]},{"id":"f2e5bfee.808d2","type":"change","z":"705e020d.9cfb5c","name":"Set topic","rules":[{"t":"set","p":"topic","pt":"msg","to":"payload.after.id","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":400,"y":180,"wires":[["65c8ad05.a4d164"]]},{"id":"38f13386.6609ec","type":"switch","z":"705e020d.9cfb5c","name":"If same event","property":"payload.after.id","propertyType":"msg","rules":[{"t":"eq","v":"$flowContext(\"frigate_event_id.\" & payload.after.camera)\t","vt":"jsonata"},{"t":"else"}],"checkall":"true","repair":false,"outputs":2,"x":780,"y":180,"wires":[["c1a0d017.a6d22"],["c5037af1.aaddb8"]]},{"id":"c1a0d017.a6d22","type":"switch","z":"705e020d.9cfb5c","name":"If score higher","property":"payload.after.top_score","propertyType":"msg","rules":[{"t":"gt","v":"$flowContext(\"frigate_top_score.\" & payload.after.camera)\t","vt":"jsonata"}],"checkall":"true","repair":false,"outputs":1,"x":1040,"y":140,"wires":[["a7172915.095e28"]]},{"id":"432fc375.b9892c","type":"server-events","z":"705e020d.9cfb5c","name":"","server":"4200f2e6.56ab5c","event_type":"telegram_callback","exposeToHomeAssistant":false,"haConfig":[{"property":"name","value":""},{"property":"icon","value":""}],"waitForRunning":true,"x":150,"y":540,"wires":[["b24fc261.ced0e"]]},{"id":"b24fc261.ced0e","type":"function","z":"705e020d.9cfb5c","name":"","func":"var full_command = msg.payload.event.command\nvar chat_id = msg.payload.event.chat_id;\nif (full_command.substr(0,4) == \"/fgv\") {\n    var event_id = full_command.substr(5);\n    var video_url = \"http://10.10.1.2:23500/clips/\" + event_id + \".mp4\";\n    return {\n        payload: {\n            chat_id: chat_id,\n            callback_id: msg.payload.event.id,\n            message_id: msg.payload.event.message.message_id,\n            event_id: event_id,\n            video_url: video_url\n        },\n        url: video_url\n    \n    };\n}\n\n","outputs":1,"noerr":0,"initialize":"","finalize":"","x":380,"y":540,"wires":[["d2e020b9.39b4d","b11d407f.1790d"]]},{"id":"3c29ef9b.57023","type":"api-call-service","z":"705e020d.9cfb5c","name":"Send video","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"send_video","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1790,"y":560,"wires":[[]]},{"id":"d2e020b9.39b4d","type":"http request","z":"705e020d.9cfb5c","name":"Check if video is available","method":"GET","ret":"txt","paytoqs":"ignore","url":"","tls":"","persist":false,"proxy":"","authType":"","x":590,"y":520,"wires":[["1b1cecba.178933"]]},{"id":"9565bcf7.c2a34","type":"join-wait","z":"705e020d.9cfb5c","name":"","paths":"[\"event\", \"check\"]","pathsToExpire":"","useRegex":false,"warnUnmatched":true,"pathTopic":"paths","pathTopicType":"msg","correlationTopic":"_msgid","correlationTopicType":"msg","timeout":"5","timeoutUnits":"60000","exactOrder":"false","firstMsg":"true","mapPayload":"true","disableComplete":false,"x":1020,"y":560,"wires":[["3d7f2f2f.c02e5"],[]]},{"id":"1b1cecba.178933","type":"change","z":"705e020d.9cfb5c","name":"Set check path","rules":[{"t":"set","p":"paths","pt":"msg","to":"check","tot":"str"},{"t":"set","p":"payload","pt":"msg","to":"statusCode","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":820,"y":520,"wires":[["9565bcf7.c2a34"]],"info":"Sets path to \"check\" to be evaluated in the join-wait node"},{"id":"b11d407f.1790d","type":"change","z":"705e020d.9cfb5c","name":"Set event path","rules":[{"t":"set","p":"paths","pt":"msg","to":"event","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":820,"y":600,"wires":[["9565bcf7.c2a34"]],"info":"Sets path to \"event\" to be evaluated in the join-wait node"},{"id":"3d7f2f2f.c02e5","type":"switch","z":"705e020d.9cfb5c","name":"If statusCode 200","property":"paths.check","propertyType":"msg","rules":[{"t":"eq","v":"200","vt":"str"},{"t":"else"}],"checkall":"true","repair":false,"outputs":2,"x":1230,"y":560,"wires":[["76fa9e49.c8318","1aa3fd8b.d744c2"],["c88c1228.857f5"]]},{"id":"e1d456db.8b2b08","type":"delay","z":"705e020d.9cfb5c","name":"","pauseType":"delay","timeout":"5","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"x":380,"y":600,"wires":[["d2e020b9.39b4d","b11d407f.1790d"]]},{"id":"c88c1228.857f5","type":"counter","z":"705e020d.9cfb5c","name":"","init":"0","step":"1","lower":"","upper":"18","mode":"increment","outputs":"1","x":1200,"y":620,"wires":[["9ae93367.c7dce"]]},{"id":"9ae93367.c7dce","type":"switch","z":"705e020d.9cfb5c","name":"Count limit reached","property":"countUpperLimitReached","propertyType":"msg","rules":[{"t":"true"},{"t":"else"}],"checkall":"true","repair":false,"outputs":2,"x":1230,"y":680,"wires":[["b6ff9d31.1f4ef"],["e1d456db.8b2b08","832faadb.f9b818"]]},{"id":"76fa9e49.c8318","type":"change","z":"705e020d.9cfb5c","name":"Event to payload","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"set","p":"payload.data.target","pt":"msg","to":"paths.event.chat_id","tot":"msg"},{"t":"set","p":"payload.data.url","pt":"msg","to":"paths.event.video_url","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1590,"y":560,"wires":[["3c29ef9b.57023"]]},{"id":"586e3369.a3626c","type":"api-call-service","z":"705e020d.9cfb5c","name":"Send not found message","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"send_message","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1830,"y":660,"wires":[[]]},{"id":"b6ff9d31.1f4ef","type":"change","z":"705e020d.9cfb5c","name":"Event to payload","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"set","p":"payload.data.target","pt":"msg","to":"paths.event.chat_id","tot":"msg"},{"t":"set","p":"payload.data.message","pt":"msg","to":"I am sorry, but the video was not found. Check it in Home Assistant","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":1590,"y":660,"wires":[["586e3369.a3626c"]]},{"id":"832faadb.f9b818","type":"switch","z":"705e020d.9cfb5c","name":"","property":"count","propertyType":"msg","rules":[{"t":"eq","v":"1","vt":"str"}],"checkall":"true","repair":false,"outputs":1,"x":1410,"y":720,"wires":[["977d864e.26e2f8"]]},{"id":"328d440f.88450c","type":"api-call-service","z":"705e020d.9cfb5c","name":"Wait notification","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"answer_callback_query","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1800,"y":720,"wires":[[]]},{"id":"977d864e.26e2f8","type":"change","z":"705e020d.9cfb5c","name":"Event to payload","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"set","p":"payload.data.callback_query_id","pt":"msg","to":"paths.event.callback_id","tot":"msg"},{"t":"set","p":"payload.data.message","pt":"msg","to":"Video not yet available. I will keep trying and let you know.","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":1590,"y":720,"wires":[["328d440f.88450c"]]},{"id":"e8379e9b.a5e46","type":"delay","z":"705e020d.9cfb5c","name":"","pauseType":"timed","timeout":"5","timeoutUnits":"seconds","rate":"1","nbRateUnits":"2","rateUnits":"minute","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":true,"x":2080,"y":220,"wires":[["b0c7e279.c2286"]]},{"id":"37ecf892.15d2b8","type":"api-call-service","z":"705e020d.9cfb5c","name":"Sending callback answer","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"answer_callback_query","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1830,"y":500,"wires":[[]]},{"id":"1aa3fd8b.d744c2","type":"change","z":"705e020d.9cfb5c","name":"Event to payload","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"set","p":"payload.data.callback_query_id","pt":"msg","to":"paths.event.callback_id","tot":"msg"},{"t":"set","p":"payload.data.message","pt":"msg","to":"Video found - sending it your way","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":1590,"y":500,"wires":[["37ecf892.15d2b8"]]},{"id":"a7172915.095e28","type":"function","z":"705e020d.9cfb5c","name":"Set new high score","func":"flow.set(\"frigate_top_score.\" + msg.payload.after.camera, msg.payload.after.top_score);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":1250,"y":140,"wires":[["8db8cd9.a79103"]]},{"id":"c5037af1.aaddb8","type":"function","z":"705e020d.9cfb5c","name":"Set last event ID","func":"flow.set(\"frigate_event_id.\" + msg.payload.after.camera, msg.payload.after.id);\nflow.set(\"frigate_top_score.\" + msg.payload.after.camera, msg.payload.after.top_score);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":1040,"y":220,"wires":[["8db8cd9.a79103"]]},{"id":"6ed0f01a.1a549","type":"change","z":"705e020d.9cfb5c","name":"Set topic to camera","rules":[{"t":"set","p":"topic","pt":"msg","to":"payload.after.camera","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1850,"y":220,"wires":[["e8379e9b.a5e46"]],"info":"This is for the next rate limit delay node. \nI want to rate-limit per camera."},{"id":"5d7fa5ed.fb763c","type":"comment","z":"705e020d.9cfb5c","name":"== Send event notification to Telegram ==","info":"","x":200,"y":60,"wires":[]},{"id":"ea0bddb3.5f172","type":"comment","z":"705e020d.9cfb5c","name":"== Process the Get Video callback and send it ==","info":"","x":220,"y":420,"wires":[]},{"id":"4ef71831.e414b8","type":"comment","z":"705e020d.9cfb5c","name":"Set your Telegram chat_id and Frigate IP","info":"","x":2380,"y":180,"wires":[]},{"id":"ed415bf5.0608d8","type":"comment","z":"705e020d.9cfb5c","name":"Set your Frigate IP in the function block","info":"","x":410,"y":480,"wires":[]},{"id":"84148b5c.b50cd8","type":"comment","z":"705e020d.9cfb5c","name":"Prepare data attributes (open this for more info)","info":"I am using change nodes to prepare the data payload.\nThis is mostly because with mustache templates directly on the call service node I couldn't find a way to put the \"targer\" or \"callback_id\" as integer and not as string (mustache seems to be supported only in double quotes) and then it complains about the ID not being int.","x":1680,"y":440,"wires":[]},{"id":"a86136e0.7fe238","type":"mqtt-broker","name":"Local","broker":"localhost","port":"1883","clientid":"","usetls":false,"compatmode":true,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"4200f2e6.56ab5c","type":"server","name":"Home Assistant","legacy":false,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":false}]
```

**Comment by InToSSH** [comment]
Here you go guys.
I have added some comments, I hope it will make sense. 
It could be made neat-er by using env vars for the settings or even putting all in a subflow, but don't have time for it now.
If you need some help, let me know.
```
[{"id":"705e020d.9cfb5c","type":"tab","label":"Frigate (export)","disabled":false,"info":""},{"id":"5d7fa5ed.fb763c","type":"comment","z":"705e020d.9cfb5c","name":"== Send event notification to Telegram ==","info":"","x":200,"y":80,"wires":[]},{"id":"ea0bddb3.5f172","type":"comment","z":"705e020d.9cfb5c","name":"== Process the Get Video callback and send it ==","info":"","x":220,"y":440,"wires":[]},{"id":"4ef71831.e414b8","type":"comment","z":"705e020d.9cfb5c","name":"Set your Telegram chat_id and Frigate IP:port","info":"","x":2190,"y":200,"wires":[]},{"id":"ed415bf5.0608d8","type":"comment","z":"705e020d.9cfb5c","name":"Set your Frigate IP and port in the function block","info":"","x":440,"y":520,"wires":[]},{"id":"8e7494e4.0e2268","type":"mqtt in","z":"705e020d.9cfb5c","name":"","topic":"frigate/events","qos":"2","datatype":"auto","broker":"a86136e0.7fe238","x":110,"y":200,"wires":[["a071a95e.e3e1c8"]]},{"id":"a071a95e.e3e1c8","type":"json","z":"705e020d.9cfb5c","name":"","property":"payload","action":"","pretty":false,"x":250,"y":200,"wires":[["77379734.89ee08"]]},{"id":"1c9e9866.1ac748","type":"api-call-service","z":"705e020d.9cfb5c","name":"","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"send_photo","entityId":"","data":"{\"target\":1111100000,\"url\":\"http://10.10.1.2:23500/events/{{payload.after.id}}/snapshot.jpg\",\"caption\":\"{{payload.after.label}} detected at {{payload.after.camera}}\",\"inline_keyboard\":[\"Get video:/fgv:{{payload.after.camera}}-{{payload.after.id}}, Siren short:/sirenshort, Siren:/siren\"]}","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":2150,"y":240,"wires":[[]]},{"id":"38ff38fc.71e138","type":"switch","z":"705e020d.9cfb5c","name":"If score >= 0.6","property":"payload.after.score","propertyType":"msg","rules":[{"t":"gte","v":"0.6","vt":"str"}],"checkall":"true","repair":false,"outputs":1,"x":580,"y":200,"wires":[["98c8ad71.82c3d"]]},{"id":"e24e3676.58d608","type":"rbe","z":"705e020d.9cfb5c","name":"Only once per event","func":"rbe","gap":"","start":"","inout":"out","property":"topic","x":1640,"y":240,"wires":[["f21815df.6eb9b8"]]},{"id":"5772ec40.9363b4","type":"trigger","z":"705e020d.9cfb5c","name":"","op1":"","op2":"","op1type":"nul","op2type":"payl","duration":"5","extend":true,"overrideDelay":false,"units":"s","reset":"","bytopic":"topic","topic":"topic","outputs":1,"x":1460,"y":240,"wires":[["e24e3676.58d608"]]},{"id":"77379734.89ee08","type":"change","z":"705e020d.9cfb5c","name":"Set topic","rules":[{"t":"set","p":"topic","pt":"msg","to":"payload.after.id","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":400,"y":200,"wires":[["38ff38fc.71e138"]]},{"id":"98c8ad71.82c3d","type":"switch","z":"705e020d.9cfb5c","name":"If same event","property":"payload.after.id","propertyType":"msg","rules":[{"t":"eq","v":"$flowContext(\"frigate_event_id.\" & payload.after.camera)\t","vt":"jsonata"},{"t":"else"}],"checkall":"true","repair":false,"outputs":2,"x":780,"y":200,"wires":[["2db72b85.631504"],["409cf951.8ef278"]]},{"id":"2db72b85.631504","type":"switch","z":"705e020d.9cfb5c","name":"If score higher","property":"payload.after.top_score","propertyType":"msg","rules":[{"t":"gt","v":"$flowContext(\"frigate_top_score.\" & payload.after.camera)\t","vt":"jsonata"}],"checkall":"true","repair":false,"outputs":1,"x":1040,"y":160,"wires":[["32a9c6e7.be4f6a"]]},{"id":"226c892f.043e46","type":"server-events","z":"705e020d.9cfb5c","name":"","server":"4200f2e6.56ab5c","event_type":"telegram_callback","exposeToHomeAssistant":false,"haConfig":[{"property":"name","value":""},{"property":"icon","value":""}],"waitForRunning":true,"x":150,"y":560,"wires":[["7ea214e3.74c13c"]]},{"id":"7ea214e3.74c13c","type":"function","z":"705e020d.9cfb5c","name":"","func":"var full_command = msg.payload.event.command\nvar chat_id = msg.payload.event.chat_id;\nif (full_command.substr(0,4) == \"/fgv\") {\n    var event_id = full_command.substr(5);\n    var video_url = \"http://10.10.1.2:23500/clips/\" + event_id + \".mp4\";\n    return {\n        orig_payload: {\n            target: chat_id,\n            callback_query_id: msg.payload.event.id,\n            message_id: msg.payload.event.message.message_id,\n            event_id: event_id,\n            url: video_url\n        },\n        url: video_url\n    \n    };\n}\n\n","outputs":1,"noerr":0,"initialize":"","finalize":"","x":380,"y":560,"wires":[["2f3408fa.9d6628"]]},{"id":"2d7078e4.502628","type":"api-call-service","z":"705e020d.9cfb5c","name":"Send video","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"send_video","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1610,"y":560,"wires":[[]]},{"id":"2f3408fa.9d6628","type":"http request","z":"705e020d.9cfb5c","name":"Check if video is available","method":"GET","ret":"txt","paytoqs":"ignore","url":"","tls":"","persist":false,"proxy":"","authType":"","x":590,"y":560,"wires":[["5a5afab0.e2bc04"]]},{"id":"5a5afab0.e2bc04","type":"change","z":"705e020d.9cfb5c","name":"Set orig payload","rules":[{"t":"delete","p":"payload","pt":"msg"},{"t":"set","p":"payload.data","pt":"msg","to":"orig_payload","tot":"msg"},{"t":"set","p":"check","pt":"msg","to":"statusCode","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":840,"y":560,"wires":[["e6a6eb27.a9a0a8"]],"info":"Sets path to \"check\" to be evaluated in the join-wait node"},{"id":"e6a6eb27.a9a0a8","type":"switch","z":"705e020d.9cfb5c","name":"If statusCode 200","property":"check","propertyType":"msg","rules":[{"t":"eq","v":"200","vt":"str"},{"t":"else"}],"checkall":"true","repair":false,"outputs":2,"x":1050,"y":560,"wires":[["da528c7c.3a9f8","2d7078e4.502628"],["7a4c12b1.5c006c"]]},{"id":"c481cef1.36807","type":"delay","z":"705e020d.9cfb5c","name":"","pauseType":"delay","timeout":"5","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"x":400,"y":620,"wires":[["2f3408fa.9d6628"]]},{"id":"7a4c12b1.5c006c","type":"counter","z":"705e020d.9cfb5c","name":"","init":"0","step":"1","lower":"","upper":"18","mode":"increment","outputs":"1","x":1020,"y":620,"wires":[["b4ea6814.9aebe8"]]},{"id":"b4ea6814.9aebe8","type":"switch","z":"705e020d.9cfb5c","name":"Count limit reached","property":"countUpperLimitReached","propertyType":"msg","rules":[{"t":"true"},{"t":"else"}],"checkall":"true","repair":false,"outputs":2,"x":1050,"y":680,"wires":[["ba27c89b.05b8a8"],["243cbdbb.ee08d2","a47ddb4e.6d81b8"]]},{"id":"43922b1b.24db54","type":"api-call-service","z":"705e020d.9cfb5c","name":"Send not found message","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"send_message","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1650,"y":660,"wires":[[]]},{"id":"ba27c89b.05b8a8","type":"change","z":"705e020d.9cfb5c","name":"Add message","rules":[{"t":"set","p":"payload.data.message","pt":"msg","to":"I am sorry, but the video was not found. Check it in Home Assistant","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":1400,"y":660,"wires":[["43922b1b.24db54"]]},{"id":"243cbdbb.ee08d2","type":"switch","z":"705e020d.9cfb5c","name":"","property":"count","propertyType":"msg","rules":[{"t":"eq","v":"1","vt":"str"}],"checkall":"true","repair":false,"outputs":1,"x":1250,"y":720,"wires":[["7fdb91a2.203f9"]]},{"id":"8b37762a.cd6098","type":"api-call-service","z":"705e020d.9cfb5c","name":"Wait notification","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"answer_callback_query","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1620,"y":720,"wires":[[]]},{"id":"7fdb91a2.203f9","type":"change","z":"705e020d.9cfb5c","name":"Add message","rules":[{"t":"set","p":"payload.data.message","pt":"msg","to":"Video not yet available. I will keep trying and let you know.","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":1400,"y":720,"wires":[["8b37762a.cd6098"]]},{"id":"9b8316df.439358","type":"api-call-service","z":"705e020d.9cfb5c","name":"Sending callback answer","server":"4200f2e6.56ab5c","version":1,"debugenabled":false,"service_domain":"telegram_bot","service":"answer_callback_query","entityId":"","data":"","dataType":"json","mergecontext":"","output_location":"","output_location_type":"none","mustacheAltTags":false,"x":1650,"y":500,"wires":[[]]},{"id":"da528c7c.3a9f8","type":"change","z":"705e020d.9cfb5c","name":"Add message","rules":[{"t":"set","p":"payload.data.message","pt":"msg","to":"Video found - sending it your way","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":1400,"y":500,"wires":[["9b8316df.439358"]]},{"id":"32a9c6e7.be4f6a","type":"function","z":"705e020d.9cfb5c","name":"Set new high score","func":"flow.set(\"frigate_top_score.\" + msg.payload.after.camera, msg.payload.after.top_score);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":1250,"y":160,"wires":[["5772ec40.9363b4"]]},{"id":"409cf951.8ef278","type":"function","z":"705e020d.9cfb5c","name":"Set last event ID","func":"flow.set(\"frigate_event_id.\" + msg.payload.after.camera, msg.payload.after.id);\nflow.set(\"frigate_top_score.\" + msg.payload.after.camera, msg.payload.after.top_score);\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","x":1040,"y":240,"wires":[["5772ec40.9363b4"]]},{"id":"a47ddb4e.6d81b8","type":"change","z":"705e020d.9cfb5c","name":"Reset payload to orig location","rules":[{"t":"set","p":"orig_payload","pt":"msg","to":"payload.data","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":170,"y":620,"wires":[["c481cef1.36807"]],"info":"Sets path to \"check\" to be evaluated in the join-wait node"},{"id":"f21815df.6eb9b8","type":"trigger","z":"705e020d.9cfb5c","name":"1msg/2min per camera","op1":"","op2":"","op1type":"pay","op2type":"nul","duration":"2","extend":false,"overrideDelay":false,"units":"min","reset":"","bytopic":"topic","topic":"payload.after.camera","outputs":1,"x":1880,"y":240,"wires":[["1c9e9866.1ac748"]]},{"id":"9ab0f385.f4da3","type":"comment","z":"705e020d.9cfb5c","name":"Optional","info":"","x":1840,"y":200,"wires":[]},{"id":"cd3b053b.384068","type":"comment","z":"705e020d.9cfb5c","name":"Answering the callback - otherwise Telegram button keeps loading","info":"","x":1770,"y":460,"wires":[]},{"id":"a86136e0.7fe238","type":"mqtt-broker","name":"Local","broker":"localhost","port":"1883","clientid":"","usetls":false,"compatmode":true,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""},{"id":"4200f2e6.56ab5c","type":"server","name":"Home Assistant","legacy":false,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":false}]
```

**Comment by HITChris** [comment]
I'm not sure how Blake's payload.before.false_positive==true + after.end_time!=null conditions wouldn't filter a given sequence down to just the first/last messages. I don't believe I'm observed other behaviors so far, using beta 3. Surely the 3-5 images that @Halytskyi is still seeing are a result of the detected object being lost + redetected (new id)?

As an aside, to make filtering events a little more straightforward, maybe we can consider adding something to the payload that broadly indicates _why_ a given update was triggered out? E.g., "event_type" = 'begin', 'score change', 'zone change', 'end', etc.

In my case, I'm using TTS to communicate an object's movement from one zone to the next ('I'm detecting a person in the street', >> '... in the driveway' >> '... on the porch'. This means passing along first/last, but also any messages containing changes between the before/after current_zones arrays. Discarding top score updates is relatively easy in Nodered, but I'm not sure how realistic something like that is in yaml

**Comment by Halytskyi** [comment]
Ok, at 1st I solved this "issue" in similar way as @InToSSH did (but with 5s delay also after sending photo and with button for best image):
```
- alias: Person detected in monitored zone
  trigger:
    platform: mqtt
    topic: frigate/events
  action:
    - delay:
        seconds: 5
    - service: script.telegram_notify_time_period_photo_inkey
      data_template:
        message: 'A {{trigger.payload_json["after"]["label"]}} was detected'
        data:
          photo:
            url: 'http://my_frigate_host/events/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg'
            caption: '<b>Attention!</b> A <b>{{trigger.payload_json["after"]["label"]}}</b> was detected on <b>{{ trigger.payload_json["after"]["camera"] }}</b> camera'
          inline_keyboard:
            - 'Best image:/frigate_best_image'
    - delay:
        seconds: 5

- alias: Send best image for 'outdoor_front' camera
  trigger:
    platform: event
    event_type: telegram_callback
    event_data:
      command: "/frigate_best_image"
  action:
    - service: telegram_bot.answer_callback_query
      data:
        callback_query_id: "{{ trigger.event.data.id }}"
        message: "Requested best image"
    - service: telegram_bot.send_photo
      data:
        target: "{{ trigger.event.data.chat_id }}"
        url: "http://my_frigate_host/outdoor_front/person/best.jpg"
```

But after that I "improved" it (removed delay) with checking event ID (via "input_text"):
input_text:
```
frigate_camera_outdoor_front_event_id:
  name: Frigate camera 'outdoor_front' event ID

frigate_camera_outdoor_front_object_name:
  name: Frigate camera 'outdoor_front' object name
```

automation:
```
- alias: Object detected in monitored zone on 'outdoor_front' camera
  trigger:
    platform: mqtt
    topic: frigate/events
  condition:
    condition: and
    conditions:
      - "{{ trigger.payload_json['after']['camera'] == 'outdoor_front' }}"
      - "{{ trigger.payload_json['after']['id'] != states('input_text.frigate_camera_outdoor_front_event_id') }}"
  action:
    - service: input_text.set_value
      data:
        entity_id: input_text.frigate_camera_outdoor_front_event_id
        value: "{{ trigger.payload_json['after']['id'] }}"
    - service: input_text.set_value
      data:
        entity_id: input_text.frigate_camera_outdoor_front_object_name
        value: "{{ trigger.payload_json['after']['label'] }}"
    - service: script.telegram_notify_time_period_photo_inkey
      data_template:
        message: "A {{ trigger.payload_json['after']['label'] }} was detected"
        data:
          photo:
            url: "http://my_frigate_host/events/{{ trigger.payload_json['after']['id'] }}/snapshot.jpg"
            caption: "<b>Attention!</b> A <b>{{ trigger.payload_json['after']['label'] }}</b> was detected on <b>{{ trigger.payload_json['after']['camera'] }}</b> camera"
          inline_keyboard:
            - "Best image:/frigate_camera_outdoor_best_image"

- alias: Send best image for 'outdoor_front' camera
  trigger:
    platform: event
    event_type: telegram_callback
    event_data:
      command: "/frigate_camera_outdoor_best_image"
  action:
    - service: telegram_bot.answer_callback_query
      data:
        callback_query_id: "{{ trigger.event.data.id }}"
        message: "Requested best image"
    - service: telegram_bot.send_photo
      data:
        target: "{{ trigger.event.data.chat_id }}"
        url: "http://my_frigate_host/outdoor_front/{{ states('input_text.frigate_camera_outdoor_front_object_name') }}/best.jpg"
```

With that implementation I immediately getting ONE image when object detected and I can get "best image" if it will be necessary. In some future with having ID in variable ("input_text") it will be possible to add buttons for getting video clip ("`/clips/<camera>-<id>.mp4`", before compress it as for my cameras it >10Mb now) and  "JPG snapshot" (when it will be released in frigate v8 "`/clips/<camera>-<id>.jpg`") for current event.
Hope these examples will be useful for someone :)

**Comment by InToSSH** [comment]
@Halytskyi nicely done directly in HA :) 
Just to clarify my Node-Red automation. It does not simply wait 5s. Those 5sec are merely to say "if you don't get a better picture in next 5sec, send the best you have now"
This means it keeps track of the event ID and listens for event updates, checks the score and waits up to 5sec to get a new better snapshot, if better snapshot arrives if replaces it and waits up to another 5sec, when no better picture is received in those 5secs it sends the best it got. This way it doesn't wait for the event to end (as it could take up to 2 minutes in my case) but also doesn't send the first picture right away and waits for somewhat good one.

But I like your approach of having a button to get the best one after the event ended. I think I will add that as a second button to my current "Get video" button. :)

**Comment by blakeblackshear** [comment]
Just pushed a change to address this directly: https://github.com/blakeblackshear/frigate/commit/14d218af465d2c769027a4260a0518482a47a99e

![image](https://media.giphy.com/media/keTQt1Ry7yM9dlwDkp/giphy.gif)
